public/index.html

```diff
@@ -4,11 +4,23 @@
   <link rel="stylesheet" type="text/css" href="styles/interface.css">
   <script src="/socket.io/socket.io.js"></script>
   <script src="scripts/main.js"></script>
+  <script src="http://code.jquery.com/jquery-1.11.0.min.js"></script>
 </head>
 
 <body>
-  <input id="input" autocomplete="off" /><button onClick="sendMessage()">SEND</button>
+
+  
   <ul id='messages'></ul>
 </body>
+<footer>
+  <input class="input" id="input" autocomplete="off" action="sendMessage()" /><button class="button" id="button" onClick="sendMessage()">SEND</button>
+  <script>
+    $("#input").keyup(function(event) {
```

Comment:
Oh shoot. I didn't actually test it out or anything. I was just intuitively thinking that down would feel snappier. Didn't think about it repeatedly firing events.

One thing to keep in mind is that we'll be able to handle the case where they hold it down eventually. We're not making a chat app, we're making a game where they can only even hit enter when they have a valid sentence. Maybe we only bind the event listener they have a valid sentence? Or we just check for validation each time they try to submit?

Whatever we end up doing, I think the repeat fire issue will be easily solvable, though I don't know whether keyup would still be better. I'll let you make the call since you've done more research